pkg/cmd/roachtest/main.go

```diff
@@ -361,8 +372,8 @@ func testShouldBeSkipped(
 		return false
 	}
 
-	td, ok := testNamesToRun[test.Name]
-	return ok && test.Skip == "" && !td.Selected
+	td := testNamesToRun[test.Name]
+	return test.Skip == "" && td != nil && !td.Selected
```

Comment:
nit: I understand why we check for the `test.Skip == ""` condition here, so we don't overwrite existing skip details. But I think it's confusing to return that a test shouldn't be skipped if the test is marked with skip. Seems like it should be the opposite to me? Maybe we can extract the `test.Skip == ""` logic one layer to when we actually do the overwriting of `test.skip`?

Reply:
Sure! I can do this!